src/F1_plotting.py

```diff
@@ -171,95 +162,220 @@ def parse_simulation_log(path_log_file=None, log_type="ERROR"):
+
+        # draw the edges between the nodes based on each bus inputs/outputs
+        for bus in self.busses:
+            for component in bus.inputs:
+                # draw an arrow from the component to the bus
+                self.connect(component, bus)
+            for component in bus.outputs:
+                # draw an arrow from the bus to the component
+                self.connect(bus, component)
+
+    def add_bus(self, label="Bus", subgraph=None):
+        if subgraph is None:
+            dot = self.dot
+        else:
+            dot = subgraph
+        dot.node(
+            label,
+            shape="rectangle",
+            fontsize="10",
+            fixedsize="shape",
+            width="4.1",
+            height="0.3",
+            style="filled",
+            color="lightgrey",
+        )
 
-    # grph = graph.create_nx_graph(
-    #    energysystem, filename=user_input[PATH_OUTPUT_FOLDER] + "network_graph.xml"
-    # )
-
-    grph = graph.create_nx_graph(energysystem)
-
-    if type(node_color) is dict:
-        node_color = [node_color.get(g, "#AFAFAF") for g in grph.nodes()]
-
-    # set drawing options
-    options = {
-        "prog": "dot",
-        "with_labels": with_labels,
-        "node_color": node_color,
-        "edge_color": edge_color,
-        "node_size": node_size,
-        "arrows": arrows,
-        "font_size": 12,
-        "font_color": "w",
-    }
-
-    # draw graph
-    fig, ax = plt.subplots(figsize=(20, 10))
-    pos = nx.drawing.nx_agraph.graphviz_layout(grph, prog=layout)
-
-    nx.draw(grph, pos=pos, ax=ax, **options)
-
-    # add edge labels for all edges
-    if edge_labels is True and plt:
-        labels = nx.get_edge_attributes(grph, "weight")
-        nx.draw_networkx_edge_labels(grph, pos=pos, edge_labels=labels)
+    def add_sink(self, label="Sink", subgraph=None):
+        if subgraph is None:
+            dot = self.dot
+        else:
+            dot = subgraph
+        dot.node(
+            fixed_width_text(label, char_num=self.txt_width),
+            shape="trapezium",
+            fontsize="10",
+        )
 
-    if show_plot is True:
-        fig.show()
+    def add_source(self, label="Source", subgraph=None):
+        if subgraph is None:
+            dot = self.dot
+        else:
+            dot = subgraph
+        dot.node(
+            fixed_width_text(label, char_num=self.txt_width),
+            shape="invtrapezium",
+            fontsize="10",
+        )
 
-    if save_plot is True:
-        path = os.path.join(user_input[PATH_OUTPUT_FOLDER], "network_graph.png")
-        dict_values[PATHS_TO_PLOTS][PLOTS_NX] += [str(path)]
+    def add_transformer(self, label="Transformer", subgraph=None):
+        if subgraph is None:
+            dot = self.dot
+        else:
+            dot = subgraph
+        dot.node(
+            fixed_width_text(label, char_num=self.txt_width),
+            shape="rectangle",
+            fontsize="10",
```

Comment:
Why dont you define the fontsize on top of the file as global variable?

Reply:
Good catch

Reply:
Done in ca2110e12316360a7607046bbfe4cef8ecc78b67